llvm/lib/Target/AMDGPU/SIWholeQuadMode.cpp

```diff
@@ -96,6 +96,9 @@ enum {
   StateStrict = StateStrictWWM | StateStrictWQM,
 };
 
+using InstRange =
```

Comment:
Could this be a standard `iterator_range` instead?

Could you do the `InstRange`-related stuff as a separate NFC patch? It's hard for me to see where the substantive changes are in this PR.

Reply:
I've rolled back some of those changes (I found it easier to follow what was going on when splitBlock was closer to the place where the split instruction was selected. But it adds too much noise to the PR.

I've also split the PR into more commits.